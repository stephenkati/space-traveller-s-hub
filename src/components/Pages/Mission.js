import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { getMissions, joinMission, leaveMission } from '../../redux/missions/MissionSlice';

const Mission = () => {
  const dispatch = useDispatch();
  const missions = useSelector((state) => state.mission.missions || []);

  useEffect(() => {
    if (!missions.length) dispatch(getMissions());
  }, [dispatch, missions.length]);

  return (
    <table className="mission-table" cellSpacing={0}>
      <thead>
        <tr>
          <th className="table-head">Mission</th>
          <th className="table-head">Description</th>
          <th className="table-head">Status</th>
          <th className="table-head">Join</th>
        </tr>
      </thead>
      <tbody>
        {
          missions.map((mission) => (
            <tr className="mission-list" key={mission.mission_id}>
              <td className="mission-name"><span>{mission.mission_name}</span></td>
              <td className="description">{mission.description}</td>
              {!mission.member && (
                <>
                  <td>
                    <span className="not-a-member">NOT A MEMBER</span>
                  </td>
                  <td>
                    <button
                      type="button"
                      onClick={() => dispatch(joinMission(mission.mission_id))}
                      className="join-mission"
                    >
                      Join Mission
                    </button>
                  </td>
                </>
              )}
              {mission.member && (
                <>
                  <td>
                    <span className="active-member">Active member</span>
                  </td>
                  <td>
                    <button
                      type="button"
                      onClick={() => dispatch(leaveMission(mission.mission_id))}
                      className="leave-mission"
                    >
                      Leave Mission
                    </button>
                  </td>
                </>
              )}
            </tr>
          ))
        }
      </tbody>
    </table>
  );
};

export default Mission;
