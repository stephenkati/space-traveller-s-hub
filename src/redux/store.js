import { configureStore } from "@reduxjs/toolkit";
import MissionReducer from "./missions/MissionSlice";

const store = configureStore({
  reducer: {
    mission: MissionReducer,
  }
});

export default store;
