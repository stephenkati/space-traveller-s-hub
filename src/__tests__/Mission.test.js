import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/store';
import Mission from '../components/Pages/Mission';

describe('tests for Missions file', () => {
  test('snapshot for missions component', () => {
    const component = render(
      <BrowserRouter>
        <Provider store={store}>
          <Mission />
        </Provider>
      </BrowserRouter>,
    );

    expect(component).toMatchSnapshot();
  });
});
