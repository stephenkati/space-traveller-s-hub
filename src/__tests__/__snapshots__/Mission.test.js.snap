// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for Missions file snapshot for missions component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <table
        cellspacing="0"
        class="mission-table"
      >
        <thead>
          <tr>
            <th
              class="table-head"
            >
              Mission
            </th>
            <th
              class="table-head"
            >
              Description
            </th>
            <th
              class="table-head"
            >
              Status
            </th>
            <th
              class="table-head"
            >
              Join
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </body>,
  "container": <div>
    <table
      cellspacing="0"
      class="mission-table"
    >
      <thead>
        <tr>
          <th
            class="table-head"
          >
            Mission
          </th>
          <th
            class="table-head"
          >
            Description
          </th>
          <th
            class="table-head"
          >
            Status
          </th>
          <th
            class="table-head"
          >
            Join
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
